libmscore/score.cpp

```diff
@@ -3950,6 +3982,44 @@ int Score::nmeasures() const
       return n;
       }
 
+//---------------------------------------------------------
+//   firstTrailingMeasure
+//---------------------------------------------------------
+
+Measure* Score::firstTrailingMeasure(ChordRest** cr)
+      {
+      Measure* firstMeasure = nullptr;
+      auto m = lastMeasure();
+
+      // No active selection: prepare first empty trailing measure of entire score
+      if (!cr)
+            for (; m && m->isFullMeasureRest(); firstMeasure = m, m = m->prevMeasure());
+      // Active selection: select full measure rest of active staff's empty trailing measure
+      else {
+            ChordRest* tempCR = nullptr;
+            while (m && (tempCR = m->first()->nextChordRest(trackZeroVoice((*cr)->track()), false))->isFullMeasureRest()) {
+                  *cr = tempCR;
+                  m = m->prevMeasure();
+                  }
+            }
+      return firstMeasure;
+      }
+
```

Comment:
The fun with this was having it double as a function for getting the score's first trailing measure, and also having it serve as setting the appropriate rest for a specific staff if there was a valid selection from the caller (Score::move()). I used a default nullptr rather than function overloading, which required a double pointer rather than a reference since references can't be null